compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/interop/UIKitView.uikit.kt

```diff
@@ -105,118 +112,156 @@ private fun Modifier.interopSemantics(enabled: Boolean, wrappingView: InteropWra
- * @param accessibilityEnabled If `true`, then the view will be visible to accessibility services.
- *
- * If this Composable is within a modifier chain that merges
- * the semantics of its children (such as `Modifier.clickable`), the merged subtree data will be ignored in favor of
- * the native UIAccessibility resolution for the view constructed by [factory]. For example, `Button` containing [UIKitView]
- * will be invisible for accessibility services, only the [UIView] created by [factory] will be accessible.
- * To avoid this behavior, set [accessibilityEnabled] to `false` and use custom [Modifier.semantics] for `Button` to
- * make the information associated with this view accessible.
- *
- * If there are multiple [UIKitView] or [UIKitViewController] with [accessibilityEnabled] set to `true` in the merged tree, only the first one will be accessible.
- * Consider using a single [UIKitView] or [UIKitViewController] with multiple views inside it if you need multiple accessible views.
- *
- * In general, [accessibilityEnabled] set to `true` is not recommended to use in such cases.
- * Consider using [Modifier.semantics] on Composable that merges its semantics instead.
- *
- * @see Modifier.semantics
+ * Internal common part of custom layout emitting a node associated with UIKit interop for [UIView] and [UIViewController].
  */
 @Composable
-fun <T : UIView> UIKitView(
-    factory: () -> T,
+private fun <T : Any> UIKitInteropLayout(
     modifier: Modifier,
-    update: (T) -> Unit = STUB_CALLBACK_WITH_RECEIVER,
-    background: Color = Color.Unspecified,
-    onRelease: (T) -> Unit = STUB_CALLBACK_WITH_RECEIVER,
-    onResize: (view: T, rect: CValue<CGRect>) -> Unit = DefaultViewResize,
-    interactive: Boolean = true,
-    accessibilityEnabled: Boolean = true,
+    update: (T) -> Unit,
+    background: Color,
+    entityHandler: InteropEntityHandler<T>,
+    onResize: (T, rect: CValue<CGRect>) -> Unit,
+    interactive: Boolean,
+    accessibilityEnabled: Boolean,
 ) {
-    // TODO: adapt UIKitView to reuse inside LazyColumn like in AndroidView:
-    //  https://developer.android.com/reference/kotlin/androidx/compose/ui/viewinterop/package-summary#AndroidView(kotlin.Function1,kotlin.Function1,androidx.compose.ui.Modifier,kotlin.Function1,kotlin.Function1)
-    val interopContainer = LocalUIKitInteropContainer.current
-    val embeddedInteropComponent = remember {
-        EmbeddedInteropView(
-            interopContainer = interopContainer,
-            onRelease
-        )
-    }
     val density = LocalDensity.current
     var rectInPixels by remember { mutableStateOf(IntRect(0, 0, 0, 0)) }
     var localToWindowOffset: IntOffset by remember { mutableStateOf(IntOffset.Zero) }
     val interopContext = LocalUIKitInteropContext.current
+    val interopContainer = LocalUIKitInteropContainer.current
 
-    EmptyLayout(
-        modifier.onGloballyPositioned { coordinates ->
+    val finalModifier = modifier
```

Comment:
It was easier without extra `val`

Reply:
Way too much indentation and it's an idiom used in the common source set as well
https://github.com/JetBrains/compose-multiplatform-core/blob/7a13310af6d609d6c8ac67bc09d0900b5d066444/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/CoreTextField.kt#L648-L669

Reply:
indentation is the same, isn't it?